crates/ruff_linter/src/rules/pep8_naming/settings.rs

```diff
@@ -84,3 +64,127 @@ impl Error for SettingsError {
+    ) -> Result<Self, SettingsError> {
+        if ignore_names.is_none() && extend_ignore_names.as_ref().map_or(true, Vec::is_empty) {
+            return Ok(IgnoreNames::Default);
+        }
+
+        let mut patterns: Vec<IdentifierPattern> = Vec::with_capacity(
+            ignore_names.as_ref().map_or(12, Vec::len)
+                + extend_ignore_names.as_ref().map_or(0, Vec::len),
+        );
+
+        // Add the ignored names from the `ignore-names` option. If omitted entirely, use the
+        // defaults
+        if let Some(names) = ignore_names {
+            for name in names {
+                patterns
+                    .push(IdentifierPattern::new(&name).map_err(SettingsError::InvalidIgnoreName)?);
+            }
+        } else {
+            patterns.extend(IgnoreNames::Default.into_vec());
+        }
+
+        // Add the ignored names from the `extend-ignore-names` option.
+        if let Some(names) = extend_ignore_names {
+            for name in names {
+                patterns
+                    .push(IdentifierPattern::new(&name).map_err(SettingsError::InvalidIgnoreName)?);
+            }
+        }
+
+        Ok(IgnoreNames::UserProvided(patterns))
+    }
+
+    /// Convert the [`IgnoreNames`] into a vector of [`IdentifierPattern`]s.
+    pub fn into_vec(self) -> Vec<IdentifierPattern> {
+        match self {
+            IgnoreNames::Default => vec![
+                IdentifierPattern::new("setUp").unwrap(),
+                IdentifierPattern::new("tearDown").unwrap(),
+                IdentifierPattern::new("setUpClass").unwrap(),
+                IdentifierPattern::new("tearDownClass").unwrap(),
+                IdentifierPattern::new("setUpModule").unwrap(),
+                IdentifierPattern::new("tearDownModule").unwrap(),
+                IdentifierPattern::new("asyncSetUp").unwrap(),
+                IdentifierPattern::new("asyncTearDown").unwrap(),
+                IdentifierPattern::new("setUpTestData").unwrap(),
+                IdentifierPattern::new("failureException").unwrap(),
+                IdentifierPattern::new("longMessage").unwrap(),
+                IdentifierPattern::new("maxDiff").unwrap(),
+            ],
+            IgnoreNames::UserProvided(patterns) => patterns,
+        }
+    }
+
+    /// Returns `true` if the given name matches any of the ignored patterns.
+    pub fn matches(&self, name: &str) -> bool {
+        match self {
+            IgnoreNames::Default => matches!(
+                name,
+                "setUp"
+                    | "tearDown"
+                    | "setUpClass"
+                    | "tearDownClass"
+                    | "setUpModule"
+                    | "tearDownModule"
+                    | "asyncSetUp"
+                    | "asyncTearDown"
+                    | "setUpTestData"
+                    | "failureException"
+                    | "longMessage"
+                    | "maxDiff"
+            ),
```

Comment:
I don't know if this is preferable to using AhoCorasick or similar. Seems like a big speed-up as-is though.

Reply:
Yeah this one is less clear to me. This time it's an equality comparison with a smaller number of strings. `AhoCorasick` might still be faster, but it'd surprise me if it was by a lot. And it wouldn't surprise me if it were slower. So I'm happy to see the improvement land. :-)